tests/PromoCodesTest.php

```diff
@@ -0,0 +1,201 @@
+{
+
+    private $organization;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['email' => 'spam+spam@keboola.com']);
+
+        foreach ($this->client->listMaintainerMembers($this->testMaintainerId) as $member) {
+            if ($member['id'] === $this->normalUser['id']) {
+                $this->client->removeUserFromMaintainer($this->testMaintainerId, $member['id']);
+            }
+
+            if ($member['id'] === $this->superAdmin['id']) {
+                $this->client->removeUserFromMaintainer($this->testMaintainerId, $member['id']);
+            }
+        }
+
+        $this->organization = $this->client->createOrganization($this->testMaintainerId, [
+            'name' => 'My org for promo codes',
+        ]);
+
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => 'spam@keboola.com']);
+        $this->client->removeUserFromOrganization($this->organization['id'], $this->superAdmin['id']);
+    }
+
+    public function testMaintainerAdminCanListAndCreatePromoCodes()
+    {
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['id' => $this->normalUser['id']]);
+        $this->normalUserClient->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
+
+        $this->normalUserClient->listPromoCodesRequest($this->testMaintainerId);
+
+        $promoCodesBeforeCreate = $this->normalUserClient->listPromoCodesRequest($this->testMaintainerId);
+        $promoCode = $this->normalUserClient->createPromoCodeRequest($this->testMaintainerId, [
+            'code' => 'TEST-' . time(),
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc6months',
+        ]);
+        $promoCodesAfterCreate = $this->normalUserClient->listPromoCodesRequest($this->testMaintainerId);
+
+        $this->assertEquals(count($promoCodesBeforeCreate) + 1, count($promoCodesAfterCreate));
+
+        $this->assertEquals($promoCode, end($promoCodesAfterCreate));
+    }
+
+    public function testCannotListPromoCodesFromRemovedOrganization()
+    {
+        $this->client->createPromoCodeRequest($this->testMaintainerId, [
+            'code' => 'TEST-' . time(),
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc6months',
+        ]);
+
+        $listBeforeRemoveOrganization = $this->client->listPromoCodesRequest($this->testMaintainerId);
+
+        $this->client->deleteOrganization($this->organization['id']);
+
+        $listAfterRemoveOrganization = $this->client->listPromoCodesRequest($this->testMaintainerId);
+
+        $this->assertLessThan(count($listBeforeRemoveOrganization), count($listAfterRemoveOrganization));
+    }
+
+    public function testDifferentOrganizationMaintainerCannotCreatePromoCodes()
```

Comment:
Tady ten test neni dobre.
Mas tam jako `projectTemplateStringId` sablonu ktera je dostupna jen superadminovi. Diky tomu ti to vraci 404 ze sablona neexistuje.

Reply:
opraveno